File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/interactive/StatedElement.scala





















// Generated on 05/03/2013 10:35:07
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait StatedElement extends TypeDeserializer {
  val typeId = 108

  protected trait Type extends BaseType with Serializable {
    val typeId = StatedElement.typeId

    val elementId: Int
    val elementCellId: Short
    val elementState: Int

    def serialize(buf: Buffer) {
      buf.writeInt(elementId)
      buf.writeShort(elementCellId)
      buf.writeInt(elementState)
    }

  }

  protected def create(elementId: Int, elementCellId: Short, elementState: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val elementId = buf.readInt()
    val elementCellId = buf.readShort()
    val elementState = buf.readInt()

    create(elementId, elementCellId, elementState)
  }

}

object StatedElement extends StatedElement {
  case class TypeImpl(elementId: Int, elementCellId: Short, elementState: Int) extends Type
  override def create(elementId: Int, elementCellId: Short, elementState: Int) = TypeImpl(elementId, elementCellId, elementState)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(elementId: Int, elementCellId: Short, elementState: Int) = create(elementId, elementCellId, elementState)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/StatedElement.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class StatedElement extends NetworkType {
    public static final short TYPE_ID = 108;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int elementId;
    public short elementCellId;
    public int elementState;
    
    public StatedElement() { }
    
    public StatedElement(int elementId, short elementCellId, int elementState) {
        this.elementId = elementId;
        this.elementCellId = elementCellId;
        this.elementState = elementState;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(elementId);
        buf.writeShort(elementCellId);
        buf.writeInt(elementState);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        elementId = buf.readInt();
        if (elementId < 0)
            throw new RuntimeException("Forbidden value on elementId = " + elementId + ", it doesn't respect the following condition : elementId < 0");
        elementCellId = buf.readShort();
        if (elementCellId < 0 || elementCellId > 559)
            throw new RuntimeException("Forbidden value on elementCellId = " + elementCellId + ", it doesn't respect the following condition : elementCellId < 0 || elementCellId > 559");
        elementState = buf.readInt();
        if (elementState < 0)
            throw new RuntimeException("Forbidden value on elementState = " + elementState + ", it doesn't respect the following condition : elementState < 0");
    }
    
}
